### PR TITLE
like and dislike feature

### PR DESCRIPTION
**Pull request by ayushkumar2808**
I have created like and dislike features and also did some bug fixes in upload.js

### CODE REVIEW

File: client/src/components/pages/Login.js [hunk 1]
```diff
@@ -19,9 +20,18 @@ const Login = () => {
   const [email,setEmail]= useState("")
  const [isLoading,setIsLoading] = useState(false)
   const navigate=useNavigate()
+  const [loading,setLoading] = useState(false)
+  useEffect(()=>{
```
Comment (aditya-190) [inline client/src/components/pages/Login.js:24]: Pls remove this useEffect and Loading. It is not good to make user wait if the data is already loaded just to show the loading animation.

File: client/src/components/pages/Projects.js [hunk 2]
```diff
@@ -1,24 +1,95 @@
...
+    fetch(process.env.REACT_APP_API+"/projects/allprojects",{
 
   }).then(res=>res.json())
   .then(result=>{
-    console.log(result.posts)
+
     setData(result.posts)
+    console.log(state)
   })
+  setTimeout(()=>{
```
Comment (aditya-190) [inline client/src/components/pages/Projects.js:26]: Pls Remove `setTimeout` and `console` here as they are not required.

File: client/src/components/pages/Upload.js [hunk 3]
```diff
@@ -9,48 +10,77 @@ const Upload = () => {
   const [link, setLink] = useState("");
   const [image, setImage] = useState("");
   const [url, setUrl] = useState("");
-  const postDetails = () => {
+  const [loading,setLoading] = useState(false)
+  useEffect(()=>{
```
Comment (aditya-190) [inline client/src/components/pages/Upload.js:14]: Pls remove `setTimeout` throughout the web app wherever it is unnecessary.

**Review by aditya-190 — CHANGES_REQUESTED** [review]
Please make the necessary changes suggested before merging.

File: client/src/components/pages/Login.js (hunk 1 above)
Comment (ayushkumar2808) [reply client/src/components/pages/Login.js:24]: can i decrease the timing of set timeout because I want to show the loader

**Pull request merged**